mods/flowers/init.lua

```diff
@@ -187,9 +177,11 @@ minetest.register_abm({
 		if not node_under then
 			return
 		end
-		if minetest.get_item_group(node_under.name, "soil") ~= 0 and
-				minetest.get_node_light(pos, nil) <= 9 and
-				minetest.get_node_light(random, nil) <= 9 then
+
+		if (minetest.get_item_group(node_under.name, "soil") ~= 0 or
+			minetest.get_item_group(node_under.name, "tree") ~= 0) and
+			minetest.get_node_light(pos, 0.5) <= 5 and
+			minetest.get_node_light(random, 0.5) <= 5 then
```

Comment:
3 lines above need indenting by one more tab to distinguish them from line 185.
The 2 light values actually need to be '<= 1' from testing in a dark area of rainforest, i did not realise rainforest could get this dark.